### ISSUE TITLE
Differentiate between bicycle=yes and bicycle=designated roads/ways

### ISSUE THREAD

**Issue by pastk** [organicmaps/organicmaps#9593]
At the moment OM treats both `bicycle=yes` and `bicycle=designated` (when applied to roads) the same: it gives a very strong preference to use such roads by the router.

It makes sense for `bicycle=designated` (usually means that the way is somehow oriented for faster, safer and more comfortable cycling).

While `bicycle=yes` shows that its legal to ride a bike there, that's it. E.g. it could be an ordinary footway or a road.

E.g. see current routing weights in https://github.com/organicmaps/organicmaps/pull/7878#issuecomment-2053733722 : `unpaved_bad yesbicycle` is considered more preferrable than `paved_good`.

**Comment by Misalf-git** [comment]
Signage like this may not be too common but familiar to me
![footway_bicycle_yes.png](https://github.com/user-attachments/assets/a25b130e-4621-483f-b332-016bbfdf2b1f)

I'd tag `highway=footway` + `bicycle=yes`.

Those may sometimes provide more than just a reasonable suggestion, but a "safe alternative" to an otherwise (maybe) "unsafe" road, where locals who know better might not even think of taking this road by bike / foot.

Like it's not a Freeway / Autobahn, therefore allowed, however, while not prohibted by law, still an unwise choice  - like a tunnel or bridge underpass where both directions cannot see in time what / who comes behind the corner.

Those would be "alternatively avoidable" (signed / "suggested") via that nearby footway, which allows cycling in whatever form for this particular situation.

**Comment by pastk** [comment]
> footway where cycling is legal, without making it a combined footway + cycleway: [highway](https://wiki.openstreetmap.org/wiki/Key:highway)=[footway](https://wiki.openstreetmap.org/wiki/Tag:highway%3Dfootway) + [bicycle](https://wiki.openstreetmap.org/wiki/Key:bicycle)=[yes](https://wiki.openstreetmap.org/wiki/Tag:bicycle%3Dyes)

IDK Germany specifics, but this looks like a sign for a shared foot & cycling way? Its explicitly designated for cyclists (in addition to pedestrians), hence should be `bicycle=designated` IMO. As its a shared (not segregated or dedicated) path then OM should give it a lesser preference than e.g. `highway=cycleway` of course.

Anyway, the original point stands:
`bicycle=yes` should have much lesser preference for routing than `bicycle=designated`.

**Comment by pastk** [comment]
And in some countries prioritizing `bicycle=yes` doesn't seem to make any sense, e.g. check examples in
- https://github.com/organicmaps/organicmaps/pull/8026

**Comment by map-per** [comment]
> IDK Germany specifics, but this looks like a sign for a shared foot & cycling way? Its explicitly designated for cyclists (in addition to pedestrians), hence should be bicycle=designated IMO.

This is going a little too much into German laws, but Footway + bicycle "frei" means that cyclists have to ride at walking speed. So using _bicycle=yes_ and not _bicycle=designated_ is a good tagging choice IMO. That means that differentiating between _yes_ and _designated_ in OM would also work as intended for these ways.

**Comment by matkoniecz** [comment]
> It makes sense for [bicycle=designated](https://wiki.openstreetmap.org/wiki/Tag:bicycle=designated) (usually means that the way is somehow oriented for faster, safer and more comfortable cycling).

I would not assume strongly this about say `highway=secondary bicycle=designated` without any tags indicating bicycle infrastructure. Sharrows may be enough for that ( https://en.wikipedia.org/wiki/Shared_lane_marking ) and this is often cargo culting without providing real infrastructure.

See say https://en.wikipedia.org/wiki/Shared_lane_marking#/media/File:Sharrows_Toronto_2011.jpg used as example picture in article.

**Comment by pastk** [comment]
Yes you're totally right that specific cycling infrastructure tags should take precedence (e.g. https://github.com/organicmaps/organicmaps/issues/5123). Still `bicycle=yes/designated` distinction is necessary for other cases.
